### PR TITLE
Clean up table functions, fix `dolt_patch` for doltgres

### PR DESCRIPTION
**Pull request by tbantle22**
Can unskip [this Doltgres test](https://github.com/dolthub/doltgresql/pull/801/files#diff-b84d6ecc91ed38b9c4d370011983d06e13a824a0a54ab3c7ec481c9e962efb40R384)

### CODE REVIEW

File: go/libraries/doltcore/sqle/database_provider.go [hunk 1]
```diff
@@ -126,6 +127,11 @@ func NewDoltDatabaseProviderWithDatabases(defaultBranch string, fs filesys.Files
 		funcs[strings.ToLower(fn.FunctionName())] = fn
 	}
 
+	tableFuncs := make(map[string]sql.TableFunction, len(dtablefunctions.DoltTableFunctions))
+	for _, fn := range dtablefunctions.DoltTableFunctions {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/database_provider.go:131]: Nice

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>9781bc7</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>9781bc7</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@tbantle22 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>33014fe</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>33014fe</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**